CDTDatastore/query/CDTQIndexManager.m

```diff
@@ -214,8 +214,31 @@ - (instancetype)initUsingDatastore:(CDTDatastore *)datastore error:(NSError *__a
  */
 - (NSString *)ensureIndexed:(NSArray * /* NSString */)fieldNames
 {
-    CDTLogError(CDTQ_LOG_CONTEXT, @"-ensureIndexed: not implemented");
-    return nil;
+    // So we don't break semvar w.r.t CDTQIndex and nullablity, we need to
```

Comment:
Well I've thought about this and there is obviously a race condition between generating the index name and then actually using it, so you could still have a collision without your code detecting it. Now of course we are only talking about a 2^160 chance of this happening. So maybe we just don't bother checking for collision?